samcli/lib/package/artifact_exporter.py

```diff
@@ -69,7 +69,7 @@ def do_export(self, resource_id, resource_dict, parent_dir):
         if (
             template_path is None
             or is_s3_url(template_path)
-            or template_path.startswith(self.uploader.s3.meta.endpoint_url)
+            or template_path.startswith(self.uploader.get(self.EXPORT_DESTINATION).s3.meta.endpoint_url)
```

Comment:
I think this must be exactly `self.uploader.s3` because `self.uploader` here is an instance of `S3Uploader` and `s3` is its property.

Reply:
https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/package/s3_uploader.py#L57

So the uploader within the dict which is assigned to s3 is S3Uploader which has the s3_client stored as a `s3` member.

Reply:
`self.uploader` is not a `dict` but `S3Uploader`, so I think this change needs to be reverted. The tests change are also wrong.

Reply:
yeah on further testing and running integration tests, seems like https://github.com/aws/aws-sam-cli/commit/0214f4b440de77785fe5723e0ec866525f6653dc is all we need.